File: CS/WpfApplication1/MyChartControl/AnotationCollection.cs
using System;
using System.Linq;
using System.ComponentModel;
using System.Collections.Generic;

namespace MyChartControl
{
    public class AnotationCollection : BindingList<Annotation>
    {

        public AnotationCollection(MyChartControl chart)
        {
            Owner = chart;
        }

        protected override void OnListChanged(ListChangedEventArgs e)
        {
            if (Owner.AnotationLayer != null)
                CreateAnotationPresenters();
            base.OnListChanged(e);
        }

        protected internal void CreateAnotationPresenters()
        {
            foreach (Annotation anotation in this)
            {
                AnotationPresenter anotationPresenter = new AnotationPresenter() { DataContext = anotation };
                Owner.AnotationLayer.Children.Add(anotationPresenter);
            }
        }
        public MyChartControl Owner { get; private set; }
    }
}

File: VB/WpfApplication1/MyChartControl/AnotationCollection.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Linq
Imports System.ComponentModel
Imports System.Collections.Generic

Namespace MyChartControl
	Public Class AnotationCollection
		Inherits BindingList(Of Annotation)

		Public Sub New(ByVal chart As MyChartControl)
			Owner = chart
		End Sub

		Protected Overrides Sub OnListChanged(ByVal e As ListChangedEventArgs)
			If Owner.AnotationLayer IsNot Nothing Then
				CreateAnotationPresenters()
			End If
			MyBase.OnListChanged(e)
		End Sub

		Protected Friend Sub CreateAnotationPresenters()
			For Each anotation As Annotation In Me
				Dim anotationPresenter As New AnotationPresenter() With {.DataContext = anotation}
				Owner.AnotationLayer.Children.Add(anotationPresenter)
			Next anotation
		End Sub
		Private privateOwner As MyChartControl
		Public Property Owner() As MyChartControl
			Get
				Return privateOwner
			End Get
			Private Set(ByVal value As MyChartControl)
				privateOwner = value
			End Set
		End Property
	End Class
End Namespace